### PR TITLE
*: support different tiflash_compute dispatch policy

### LINKED ISSUE #41783: Support dispatch mpp task using RoundRobin in disaggregated tiflash mode
## Enhancement
For now, we use consistent hash to dispatch mpp task to tiflash_compute nodes. But we found we have load balancing problem: different tiflash_compute has very different number of regions to read from storage node.

As a backup, we support roundrobin to dispatch.

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #41783

Problem Summary: As issue said.

### What is changed and how it works?
1. add `tiflash_compute_dispatch_policy` variable(session and global) to decide which policy to use.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by XuHuaiyu — APPROVED** [review]

File: kv/mpp.go [hunk 1]
```diff
@@ -150,7 +150,7 @@ type MPPDispatchRequest struct {
 type MPPClient interface {
 	// ConstructMPPTasks schedules task for a plan fragment.
 	// TODO:: This interface will be refined after we support more executors.
-	ConstructMPPTasks(context.Context, *MPPBuildTasksRequest, time.Duration) ([]MPPTaskMeta, error)
+	ConstructMPPTasks(context.Context, *MPPBuildTasksRequest, time.Duration, int) ([]MPPTaskMeta, error)
```
Comment (AilinKid) [inline kv/mpp.go:153]: type policy int
use the policy instead

File: executor/tiflashtest/tiflash_test.go [hunk 2]
```diff
@@ -1453,6 +1453,78 @@ func TestMPPMemoryTracker(t *testing.T) {
...
+			if useAS {
+				// Expect error, because TestAutoScaler return empty topo.
+				require.Contains(t, err.Error(), "Cannot find proper topo from AutoScaler")
+			} else {
+				// This error message means we use PD instead of AutoScaler.
+				require.Contains(t, err.Error(), "tiflash_compute node is unavailable")
+			}
+		}
+	}
+	failpoint.Disable("github.com/pingcap/tidb/store/copr/testWhichDispatchPolicy")
```
Comment (AilinKid) [inline executor/tiflashtest/tiflash_test.go:1525]: use defer func to move it before for loop, once require failed, it failpoint path can still be cleaned

File: util/tiflashcompute/dispatch_policy.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tiflashcompute
+
+import (
+	"github.com/pingcap/errors"
+)
+
+// Different policy to dispatching task to different tiflash_compute nods.
```
Comment (AilinKid) [inline util/tiflashcompute/dispatch_policy.go:21]: use the typed plicy type
DispatchPolicyRR policy = iota

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by AilinKid — APPROVED** [review]

**Pull request merged**